storage/storage.go

```diff
@@ -35,3 +52,14 @@ func (f Fanout) Append(s *model.Sample) {
 		a.Append(s)
 	}
 }
+
+// Throttled returns true if at least one of the SampleAppenders in the Fanout
+// slice is throttled.
+func (f Fanout) Throttled() bool {
```

Comment:
I'd express the contract as follows:

There are "best effort" appenders like the various remote storages, that will drop samples at will if they feel like it. They always return `false` on `NeedsThrottling()`. Then there are "guaranteed" appenders that will always append what they are asked for (and return an error otherwise once https://github.com/prometheus/prometheus/issues/1356 is fixed). They will use `NeedsThrottling()` to communicate that they need throttling. At the moment, we only have one "guaranteed" appender, the local storage. I cannot think of a reason to have more, so the discussion is somehow academic. But if you asked me, I'd say the whole reason to use a "guaranteed" appender is that we want all the samples there. So even if only a single one of them needs throttling, we will throttle.

> On Append() it should then only call Append() on those sub-appenders that are not busy.

This is impossible to implement. The Append() call is ignorant of where it is coming from (_und das ist auch gut so_). So for how long should it not propagate Append() calls to the overloaded appender?

In summary, I think the currently implemented logic is correct, and another logic cannot be implemented without changing the design more fundamentally.